xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-api/src/main/java/org/xwiki/search/solr/internal/metadata/DocumentSolrMetadataExtractor.java

```diff
@@ -69,6 +72,15 @@
     @Inject
     private EntityReferenceSerializer<String> entityReferenceSerializer;
 
+    @Inject
+    private UserReferenceSerializer<String> userReferenceSerializer;
+
+    // TODO: relying on this serializer should be prevented by properly rewriting XWiki#getPlainUserName
+    //  to use UserReference
+    @Inject
+    @Named("document")
+    private UserReferenceSerializer<DocumentReference> documentReferenceUserReferenceSerializer;
+
```

Comment:
I don't understand why you don't use the `ConfiguredUserResourceReferenceSerializer` (which is using the default hint).

Reply:
ah you're not using `DocumentUserResourceReferenceSerializer`(thought you were)

Reply:
Actually I understand, you're using both ATM